### PR TITLE
[tests-only][full-ci]Added tests to lists shared by me after sharee is deleted

### PR DESCRIPTION
**Pull request by SagarGi**
### Description
This PR adds test coverage for `sharedByMe` after sharee gets deleted with clearing up the user cache and the group cache for the deleted sharee and the group respectively.

```feature
Scenario: user lists shared resources for deleted sharee
Scenario: user lists shared resources for deleted group
```
### Related Issue:
https://github.com/owncloud/ocis/issues/8111

### CODE REVIEW

File: tests/acceptance/features/apiSharingNg/sharedByMe.feature [hunk 1]
```diff
@@ -1451,3 +1451,66 @@ Feature: resources shared by user
       }
     }
     """
+
+  @env-config
+  Scenario: user lists shared resources for deleted sharee
+    Given the config "GRAPH_SPACES_USERS_CACHE_TTL" has been set to "1"
```
Comment (amrita-shrestha) [inline tests/acceptance/features/apiSharingNg/sharedByMe.feature:1457]: what this step is for? what willl be then step without this step

Comment (SagarGi) [reply tests/acceptance/features/apiSharingNg/sharedByMe.feature:1457]: the `env` is for how long to keep of the user cache. By default its 60 sec. But we cannot wait for 60 sec in  a test so keeping it least which is one to delete the users cache. When it is not used the in the list shared by a sharer, even after the sharee is deleted, its details are in the response which it should not be.

**Review by amrita-shrestha — APPROVED** [review]
lgtm

File: tests/acceptance/features/bootstrap/GraphContext.php [hunk 2]
```diff
@@ -2526,6 +2526,27 @@ public function userListsTheResourcesSharedWithThemUsingGraphApi(string $user):
...
+	 * @param string $user
+	 * @param bool $isCacheClear
+	 *
+	 * @return void
+	 */
+	public function userListsResourcesSharedByHimOrHer(string $user, bool $isCacheClear = false) {
+		$credentials = $this->getAdminOrUserCredentials($user);
+		if ($isCacheClear) {
+			sleep(1);
+		}
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/GraphContext.php:2540]: would be great to have a note here mentioning `GRAPH_SPACES_GROUPS_CACHE_TTL`

File: tests/acceptance/features/bootstrap/GraphContext.php [hunk 3]
```diff
@@ -2526,6 +2526,27 @@ public function userListsTheResourcesSharedWithThemUsingGraphApi(string $user):
 		);
 	}
 
+	/**
+	 *
+	 * @param string $user
+	 * @param bool $isCacheClear
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/GraphContext.php:2532]:
```suggestion
	 * @param bool $waitForCacheExpiry
```

File: tests/acceptance/features/bootstrap/GraphContext.php [hunk 4]
```diff
@@ -2526,6 +2526,27 @@ public function userListsTheResourcesSharedWithThemUsingGraphApi(string $user):
...
+	 *
+	 * @return void
+	 */
+	public function userListsResourcesSharedByHimOrHer(string $user, bool $isCacheClear = false) {
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/GraphContext.php:2536]:
```suggestion
	public function listSharesSharedByMe(string $user, bool $isCacheClear = false) {
```

**Review by saw-jan — CHANGES_REQUESTED** [review]

**Review by saw-jan — APPROVED** [review]
:+1:

**Pull request merged**